### PR TITLE
Fix bug of query builder

### PR DESCRIPTION
**Pull request by zhang-hua**
* Fix input validation of null id property
* Fix `IN` clause
* Add more Unit Tests

# Type of change? 

- [x] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 

...

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by zhang-hua** [comment]
Also fix issue #71

File: Services.Test/helpers/QueryBuilderTest.cs [hunk 1]
```diff
@@ -18,31 +18,67 @@ public void GetDocumentsSqlTest()
...
+            Assert.Equal("chiller-02.0", querySpec.Parameters[4].Value);
+            Assert.Equal("rule.id", querySpec.Parameters[5].Value);
+            Assert.Equal("bef978d4-54f6-429f-bda5-db2494b833ef", querySpec.Parameters[6].Value);
+            Assert.Equal("device.msg.received", querySpec.Parameters[7].Value);
+            Assert.Equal("device.msg.received", querySpec.Parameters[8].Value);
+            Assert.Equal("device.msg.received", querySpec.Parameters[9].Value);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void GetDocumentsSqlWithNullIdPropertyTest()
```
Comment (jillcary) [inline Services.Test/helpers/QueryBuilderTest.cs]: Nit: Naming -- Say what the test should do if there is a nullId in the test name.

File: Services/Helpers/QueryBuilder.cs [hunk 2]
```diff
@@ -169,5 +183,10 @@ private static void ValidateInput(ref string input)
                 throw new InvalidInputException($"Input '{input}' contains invalid characters.");
             }
         }
+
+        private static IEnumerable<SqlParameter> ConvertToSqlParameters(string name, string[] values)
```
Comment (jillcary) [inline Services/Helpers/QueryBuilder.cs]: nit: could you add a comment explaining why the conversion is needed?

Comment (zhang-hua) [reply Services/Helpers/QueryBuilder.cs]: yep. good suggestion.

File: Services.Test/helpers/QueryBuilderTest.cs (hunk 1 above)
Comment (zhang-hua) [reply Services.Test/helpers/QueryBuilderTest.cs]: make sense.

**Pull request merged**